common/txmgr/types/tx_store.go

```diff
@@ -58,7 +58,7 @@ type TxStore[
 	GetInProgressEthTxAttempts(ctx context.Context, address ADDR, chainID CHAIN_ID) (attempts []TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE, ADD], err error)
 	HasInProgressTransaction(account ADDR, chainID CHAIN_ID, qopts ...pg.QOpt) (exists bool, err error)
 	// InsertEthReceipt only used in tests. Use SaveFetchedReceipts instead
-	InsertEthReceipt(receipt *Receipt[R, TX_HASH, BLOCK_HASH]) error
+	InsertEthReceipt(receipt R) (receiptID int64, err error)
```

Comment:
If this is only used in tests ideally we don't export it as part of the public interface, but can we at least suffix with `XXXTestOnly` as this is what we use to mark situations like these.

Reply:
removed it from the public interface and it's wrapped into `TestEvmTxStore` 😄